src/lib.rs

```diff
@@ -321,13 +370,10 @@ impl<T> Entry<T> {
 
     #[inline]
     fn remove(&mut self, nxt: usize) -> Option<T> {
-        if self.in_use() {
-            return match mem::replace(self, Entry::Empty(nxt)) {
-                Entry::Filled(val) => Some(val),
-                Entry::Empty(_) => unreachable!(),
-            }
+        match mem::replace(self, Entry::Empty(nxt)) {
+            Entry::Filled(val) => Some(val),
+            Entry::Empty(_) => None
         }
-        None
     }
```

Comment:
+1 test case everything! Thanks. Yes, that does look like it would cause a leak. But wouldn't this also be the case even as the current code? Considering the nxt is then ignored.

I'm still in call for using abstractions to reduce verbosity in the whole code base. Notably; collections does a really good job for handling abstractions and is *VERY* clean and easy to understand and extend.